components/scream/src/share/grid/abstract_grid.hpp

```diff
@@ -127,12 +102,24 @@ class AbstractGrid : public ekat::enable_shared_from_this<AbstractGrid>
   const lid_to_idx_map_type& get_lid_to_idx_map () const;
 
   // Set/get geometric views. The setter is virtual, so each grid can check if "name" is supported.
-  virtual void set_geometry_data (const std::string& name, const geo_view_type& data) = 0;
+  void set_geometry_data (const std::string& name, const geo_view_type& data);
   const geo_view_type& get_geometry_data (const std::string& name) const;
 
 protected:
 
-  void set_unique_grid (const std::shared_ptr<const AbstractGrid>& unique_grid);
+  // Derived classes can override these methods, which are called inside the
+  // set_dofs and set_lid_to_idx_map methods respectively, to verify that the
+  // views have been set to something that satisfies any requirement of the grid type.
+  // This class already checks the extents of the view, but derived classes can add
+  // some extra consistency check.
+  // NOTE: these methods are called *at the end* of the respective set_* method,
+  //       so that other methods can be called (e.g., get_global_max_dof_gid()).
+  virtual void check_dofs_list () const {}
```

Comment:
Do you have a one-liner that describes what this checks, and what it does in case the check passes/fails?

Reply:
Also, I don't know where the conversation in the C++ community is regarding the use of exceptions as super-fancy `if` tests, but if we want to programmatically check things and correct them if they're deficient, I would prefer to have this return a boolean that indicates whether the check passed or failed. If, on the other hand, failing this check is always a fatal or uncorrectable error, the exception is more appropriate

Reply:
That's a good point. I think I can easily turn them into methods returning bool's. The only place that uses them will always throw if dofs are bad, but returning true/false is definitely more flexible.